components/purchase_modal/purchase_modal.tsx

```diff
@@ -184,7 +201,229 @@ function Card(props: CardProps) {
+            setPriceDifference((props.monthlyPrice - props.yearlyPrice) * numValue);
+        }
+    };
+
+    const updateDisplayPage = (isMonthly: boolean) => {
+        setIsMonthly(isMonthly);
+        if (isMonthly) {
+            setDisplayPrice(props.monthlyPrice);
+        } else {
+            setDisplayPrice(props.yearlyPrice);
+        }
+    };
+
+    const userCountTooltip = (
+        <Tooltip
+            id='userCount__tooltip'
+            className='userCountTooltip'
+        >
+            <div className='tooltipTitle'>
+                <FormattedMessage
+                    defaultMessage={'Current User Count'}
+                    id={'admin.billing.subscription.userCount.tooltipTitle'}
+                />
+            </div>
+            <div className='tooltipText'>
+                <FormattedMessage
+                    defaultMessage={'You must purchase at least the current number of active users.'}
+                    id={'admin.billing.subscription.userCount.tooltipText'}
+                />
+            </div>
+        </Tooltip>
+    );
+
+    const monthlyPlan = (
+        <>
+            <div className='enable_annual_sub'>
+                <button
+                    className={'plan_action_btn ' + props.buttonDetails.customClass}
+                    disabled={props.buttonDetails.disabled}
+                    onClick={props.buttonDetails.action}
+                >{props.buttonDetails.text}</button>
+            </div>
+            <div className='plan_billing_cycle'>
+                <FormattedMessage
+                    defaultMessage={'Payment begins: {beginDate}. '}
+                    id={'admin.billing.subscription.paymentBegins'}
+                    values={{
+                        beginDate: getNextBillingDate(),
+                    }}
+                />
+                <FormattedMessage
+                    defaultMessage={
+                        'Your bill is calculated at the end of the billing cycle based on the number of enabled users. '
+                    }
+                    id={'admin.billing.subscription.freeTrialDisclaimer'}
+                />
+                <a
+                    onClick={seeHowBillingWorks}
+                >
+                    <FormattedMessage
+                        defaultMessage={'See how billing works.'}
+                        id={'admin.billing.subscription.howItWorks'}
+                    />
+                </a>
+            </div>
+        </>
+    );
+
+    const tooFewUsersErrorMessage = formatMessage({
+        id: 'admin.billing.subscription.userCount.minError',
+        defaultMessage: `Your workspace currently has ${props.usersCount} users`,
+    }, {num: props.usersCount});
+
+    const tooManyUsersErrorMessage = formatMessage({
+        id: 'admin.billing.subscription.userCount.maxError',
+        defaultMessage: `Your workspace only supports up to ${Constants.MAX_PURCHASE_SEATS} users`,
```

Comment:
Just noticed this since I'll be using some of this code for self-hosted purchases: the defaultMessage is supposed to be an exact match of the string in found in the `en.json` file, e.g. `Your workspace currently has {num} users` and `Your workspace only supports up to {num} users`. The react-intl lib tries to inject the data into the string if the translation doesn't exist (e.g. someone may not be using English, and the translators for the language they are using having gotten to this string yet), and it won't be able to. Not sure if that will be an error from react intl or if it will fail silently.